static/d3/d3_graph.js

```diff
@@ -0,0 +1,676 @@
+$( window ).load(function() {
+  $("#graph1").hide();
+  $("#graph1").css("background-color", "white");
+});
+
+function change_graph() {
+
+	var sel_value = $("#graph_option").val();
+
+	if (sel_value == "d3") {
+		$(".graph_container").css("display", "none"); 
+		$("#graph1").css("display", "block"); 		
+
+	}
+	else if (sel_value == "flot") {
+		
+		$("#graph1").css("display", "none"); 
+		$(".graph_container").css("display", "block"); 
+	
+	}
+	
+}
+
+
+function zip(a, b) {
+
+return a.map(function (e, i) {
+    return [e, b[e]];
+});
+
+
+}
+
+
+function get_max_xy(series) {
+
+  max_x = Math.max.apply(null, series.map(function(x) {return Math.max.apply(null, x.data.map(function(y){return y[0];   })); }));
+ 
+  max_y = Math.max.apply(null, series.map(function(x) {return Math.max.apply(null, x.data.map(function(y){return y[1];
+})); })); 
+ 
+  var xy = [max_x, max_y]; 
+  return xy; 
+
+}
+
+function get_xs(series, args=null) {
+
+var my_set = new Set( 
+([].concat.apply([], series.map(function (d) { return d.data; }))).map(function(xy) {return xy[0]; } 
+)); 
+return my_set; 
+
+}
+
+/* Plugin for jQuery for working with colors.
```

Comment:
Would it be worth having this in a separate file, so we know which bits this comment refers to?